integration-test/src/main/java/uk/gov/gchq/gaffer/integration/generators/BasicElementGenerator.java

```diff
@@ -36,10 +36,13 @@ public class BasicElementGenerator implements OneToOneElementGenerator<DomainObj
 
     @Override
     public Element _apply(final DomainObject domainObject) {
+        Element result = null;
         if (domainObject instanceof EntityDomainObject) {
-            return entityGenerator._apply((EntityDomainObject) domainObject);
+            result = entityGenerator._apply((EntityDomainObject) domainObject);
+        } else if (domainObject instanceof EdgeDomainObject) {
+            result = edgeGenerator._apply((EdgeDomainObject) domainObject);
         }
 
-        return edgeGenerator._apply((EdgeDomainObject) domainObject);
+        return result;
```

Comment:
Maybe we should throw if we get here instead of return null

Reply:
We could, it I think it would make the flow nicer as well. Do you know of a suitable Gaffer specific exception to use?

Reply:
There is `GafferRuntimeException` but not sure what would be best to use